src/Language/PureScript/AST/Declarations.hs

```diff
@@ -822,9 +822,9 @@ newtype AssocList k t = AssocList { runAssocList :: [(k, t)] }
   deriving (Show, Eq, Ord, Foldable, Functor, Traversable)
 
 $(deriveJSON (defaultOptions { sumEncoding = ObjectWithSingleField }) ''NameSource)
+$(deriveJSON (defaultOptions { sumEncoding = ObjectWithSingleField }) ''ExportSource)
```

Comment:
This (and a few other JSON instance rearrangements) were necessary because any instances defined below a TH `deriveJSON` call like this one won't be visible to that call, so if a data type `A` references another data type `B` in its definition, then the `deriveJSON` for `B` has to come before the one for `A`.